sycl/include/sycl/ext/oneapi/bindless_images.hpp

```diff
@@ -609,6 +609,25 @@ template <typename CoordT> constexpr size_t coord_size() {
     return CoordT::size();
   }
 }
+
+// bit_cast Color to a type the NVPTX backend is known to accept
+template <typename DataT> constexpr auto convert_color_nvptx(DataT Color) {
+  if constexpr (sizeof(DataT) == 1) {
+    return sycl::bit_cast<uint8_t>(Color);
+  } else if constexpr (sizeof(DataT) == 2) {
+    return sycl::bit_cast<uint16_t>(Color);
+  } else if constexpr (sizeof(DataT) == 4) {
+    return sycl::bit_cast<uint32_t>(Color);
+  } else if constexpr (sizeof(DataT) == 8) {
+    return sycl::bit_cast<sycl::vec<uint32_t, 2>>(Color);
+  } else if constexpr (sizeof(DataT) == 16) {
+    return sycl::bit_cast<sycl::vec<uint32_t, 4>>(Color);
+  } else {
+    assert(false);
+    return Color;
```

Comment:
Fixed.

I've moved the `static_assert` for the `sizeof(DataT)` from `write_image` to `convert_color_nvptx` as it will be used in future variants of `write_image` (layered image writes).